typings/index.d.ts

```diff
@@ -2310,47 +2312,86 @@ declare module 'discord.js' {
     public valueOf(): Collection<K, Holds>;
   }
 
-  export class ApplicationCommandManager extends BaseManager<
-    Snowflake,
-    ApplicationCommand,
-    ApplicationCommandResolvable
-  > {
+  export class ApplicationCommandManager<
+    ApplicationCommandType = ApplicationCommand<{ guild: GuildResolvable }>,
+  > extends BaseManager<Snowflake, ApplicationCommandType, ApplicationCommandResolvable> {
     constructor(client: Client, iterable?: Iterable<any>);
+    public permissions: ApplicationCommandPermissionsManager<
+      { guild: GuildResolvable; command?: ApplicationCommandResolvable },
+      { guild: GuildResolvable; command: ApplicationCommandResolvable },
+      null,
```

Comment:
This is currently breaking assignability from the deriving class (`GuildApplicationCommandManager`).

While this change is technically less specific, compiling djs without errors is something we probably want 😅 
```suggestion
      Guild | null,
```

Reply:
Fixed it in a slightly different way since there was actually an error in the first set of lines too